mlir/include/air/Dialect/AIR/AIR.td

```diff
@@ -174,6 +174,7 @@ def air_HerdOp : air_Op<"herd", [air_AsyncOpInterface,
                                  AffineScope,
                                  SingleBlockImplicitTerminator<"HerdTerminatorOp">]>,
                         Arguments<(ins OptionalAttr<SymbolNameAttr>:$sym_name,
+                                       OptionalAttr<StrAttr>:$link_with,
```

Comment:
Was this required to make the python work? Or just a convenience?

Reply:
If the `link_with` attribute is set in `air.herd`, then when aie cores are outlined in `air-to-aie`, every core of that herd shall link with the object file. https://github.com/Xilinx/mlir-air/blob/2754dcf53e9dce76df4e1a8d5f34bcb2acd1871e/mlir/lib/Conversion/AIRToAIEPass.cpp#L259

Reply:
So my understanding is it's a convenience...

Reply:
> If the `link_with` attribute is set in `air.herd`, then when aie cores are outlined in `air-to-aie`, every core of that herd shall link with the object file.
> 
> https://github.com/Xilinx/mlir-air/blob/2754dcf53e9dce76df4e1a8d5f34bcb2acd1871e/mlir/lib/Conversion/AIRToAIEPass.cpp#L259

Yeah I implemented it so I'm familiar with how it works. My question was regarding the addition of the attribute in the tablegen description, and if it is necessary to make the python work.

Reply:
The python did not work without this change. I'm sure there may be other ways to signal that there needs to be a `link_with` attribute, but this is one path that seemed reasonable (to me, at least).

It did occur to me (and I did try this) that maybe I could just specify a type of `link_with` attribute when I declare the external function. I tried several modification to the `external_function` helper but was unable to make this work (I think I was able to set it as an attribute correctly, but this attribute then was not percolated to the herd/whoever was calling this function).